src/mlpack/methods/ann/layer/softshrink.hpp

```diff
@@ -0,0 +1,238 @@
+   * 		    coefficient 'a' which is one of the training parameters.
+   * 		    Default value of lambda is 0.5.
+   */
+  SoftShrink(const double lambda = 0.5);
+
+  /**
+   * Ordinary feed forward pass of a neural network, evaluating the function
+   * f(x) by propagating the activity forward through f.
+   * 
+   * @param input Input data used for evaluating the Soft Shrink function.
+   * @param output Resulting output activation
+   */
+  template<typename InputType, typename OutputType>
+  void Forward(const InputType&& input, OutputType&& output);
+
+  /**
+   * Ordinary feed backward pass of a neural network, calculating the function
+   * f(x) by propagating x backwards through f. Using the results from the feed
+   * forward pass.
+   * 
+   * @param input The propagated input activation f(x).
+   * @param gy The backpropagated error.
+   * @param g The calculated gradient
+   */
+  template<typename DataType>
+  void Backward(const DataType&& input,
+                DataType&& gy,
+                DataType&& g);
+
+  //! Get the output parameter.
+  OutputDataType const& OutputParameter() const { return outputParameter; }
+  //! Modify the output parameter.
+  OutputDataType& OutputParameter() { return outputParameter; }
+
+  //! Get the delta.
+  OutputDataType const& Delta() const { return delta; }
+  //! Modify the delta.
+  OutputDataType& Delta() { return delta; }
+
+  //! Get the hyperparameter lambda.
+  double const& Lambda() const { return lambda; }
+  //! Modify the hyperparameter lambda.
+  double& Lambda() { return lambda; }
+
+  /**
+   * Serialize the layer.
+   */
+  template<typename Archive>
+  void serialize(Archive& ar, const unsigned int /* version */);
+
+ private:
+  /**
+   * Computes the value of Soft Shrink activation function.
+   *
+   * @param x Input data.
+   * @return f(x) 
+   */
+  double Fn(const double x)
+  {
+    if (x > lambda)
+    {
+      return x - lambda;
+    }
+    else if (x < -lambda)
+    {
+      return x + lambda;
+    }
+    return 0.;
+  }
+
+  /**
+   * Computes the value of Soft Shrink activation function using a dense matrix
+   * as input.
+   * 
+   * @param x Input data.
+   * @param y The resulting output activation.
+   */
+  template<typename eT>
+  void Fn(const arma::Mat<eT>& x, arma::Mat<eT>& y)
+  {
+    y.set_size(arma::size(x));
+    for (size_t i = 0; i < x.n_elem; i++)
+    {
+      y(i) = Fn(x(i));
+    }
+  }
+
+  /**
+   * Computes the inverse of the Soft Shrink function.
+   *
+   * @f{eqnarray*}{
+   * f^{-1}(y) &=& \left\{
+   *       \begin{array}{lr}
+   *         y + lambda & : y > 0 \\
+   *         y + lambda & : y < 0 \\
+   *         0 & : otherwise \\
+   *       \end{array}
+   * \right.
+   * @f}
+   
+   * @param y
+   * @return f^{-1}(y)
+   */
+  double Inv(const double y)
+  {
+    if ( y > 0)
+    {
+      return y + lambda;
+    }
+    else if (y < 0)
+    {
+      return y - lambda;
+    }
+    return 0;
+  }
```

Comment:
No need of curly braces for **single statements** followed by if/else etc.
Please refer [Design Guidelines](https://github.com/mlpack/mlpack/wiki/DesignGuidelines)

Reply:
Hi @Hemal-Mamtora 
I am not sure about this here. As I can see in hard_sigmoid_function.hpp the for has been used with curly braces in line 91 although which is of single line, same thing as here.
Also I have been told to use so for clean code. Although we can always consult a member for this.
Hope it helps.
Thanks for noticing.

Reply:
For single statements we don't necessary have to use statements, if you like to remove it here, please feel free; either is fine for me.